keps/sig-network/5343-kube-proxy-backends/README.md

```diff
@@ -0,0 +1,824 @@
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+With [KEP-3866] going GA, there are now 4 supported kube-proxy
+backends in-tree (`iptables`, `ipvs`, `nftables`, and `winkernel`).
+The nftables KEP noted that, as future work, we should figure if/how
+we can deprecate either or both of the older Linux backends. There has
+also been sporadic discussion about deprecating the `winkernel`
+backend or moving it out of tree, since it is not clear that it
+benefits from being in-tree.
+
+Meanwhile, an earlier KEP, [KEP-3786], had proposed moving most of the
+shared kube-proxy code into the kube-proxy staging repo
+(`k8s.io/kube-proxy`) so that it could be more easily shared by
+out-of-tree service proxy implementations. That KEP was never merged,
+though there was general agreement with the high-level idea.
+
+The exact goals of this KEP are not finalized, but in general, the
+plan is to figure out what to do with kube-proxy backends.
+
+[KEP-3866]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-network/3866-nftables-proxy/README.md
+[KEP-3786]: https://github.com/kubernetes/enhancements/pull/3788
+
+## Motivation
+
+### Goals
+
+(These are kind of "Meta-Goals", since we still need to decide exactly
+what the Goals of the KEP are.)
+
+- Figure out the situation around "the default kube-proxy backend".
+
+- Figure out a plan for deprecating the `ipvs` backend.
+
+- Figure out (in conjunction with SIG Windows) what we want to do with
+  the `winkernel` backend.
+
+Assuming we end up deciding we want to move any proxy implementation
+out of tree, we will likely want to:
+
+- Refactor and redesign the kube-proxy "library" code
+  (`k8s.io/kubernetes/pkg/proxy`) to make it more usable and more
+  maintainable, and to fix existing bugs.
+
+- Move most of `k8s.io/kubernetes/pkg/proxy/` (except for the backend
+  implementations) into the kube-proxy staging repo,
+  `k8s.io/kube-proxy`, so it can be shared with out-of-tree backends.
```

Comment:
This is the use case I'm skeptical, https://github.com/kubernetes/enhancements/pull/5344/files#r2297360157 , at this point in time of the history I very much doubt this will be consumed by someone, specially for projects that don't use nftables/iptables and use ovs or ebpf instead

Reply:
This was answered here https://github.com/kubernetes/enhancements/pull/5344#discussion_r2298122404